src/Microsoft.AspNetCore.Blazor.Server/Builder/ServerSideBlazorApplicationBuilderExtensions.cs

```diff
@@ -20,94 +19,57 @@ public static class ServerSideBlazorApplicationBuilderExtensions
         /// <param name="builder">The <see cref="IApplicationBuilder"/>.</param>
         /// <typeparam name="TStartup">A Blazor startup type.</typeparam>
         /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
-        public static IApplicationBuilder UseServerSideBlazor<TStartup>(this IApplicationBuilder builder)
+        public static IApplicationBuilder UseServerSideBlazor<TStartup>(
+            this IApplicationBuilder builder)
         {
             if (builder == null)
             {
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            return UseServerSideBlazor(builder, typeof(TStartup));
-        }
-
-        /// <summary>
-        /// Registers Server-Side Blazor in the pipeline.
-        /// </summary>
-        /// <param name="builder">The <see cref="IApplicationBuilder"/>.</param>
-        /// <param name="startupType">A Blazor startup type.</param>
-        /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
-        public static IApplicationBuilder UseServerSideBlazor(
-            this IApplicationBuilder builder,
-            Type startupType)
-        {
-            if (builder == null)
-            {
-                throw new ArgumentNullException(nameof(builder));
-            }
+            // WARNING: Don't add extra setup logic here. It's important for
+            // UseServerSideBlazor just to be shorthand for UseSignalR+UseBlazor,
+            // so that people who want to call those two manually instead can
+            // also do so. That's needed for people using Azure SignalR.
 
-            if (startupType == null)
-            {
-                throw new ArgumentNullException(nameof(startupType));
-            }
-
-            var startup = builder.ApplicationServices.GetRequiredService(startupType);
-            var wrapper = new ConventionBasedStartup(startup);
-            Action<IBlazorApplicationBuilder> configure = (b) =>
-            {
-                wrapper.Configure(b, b.Services);
-            };
-
-            UseServerSideBlazorCore(builder, configure);
-
-            builder.UseBlazor(new BlazorOptions()
-            {
-                ClientAssemblyPath = startupType.Assembly.Location,
-            });
-
-            return builder;
+            // TODO: Also allow configuring the endpoint path.
+            return UseSignalR(builder, BlazorHub.DefaultPath)
```

Comment:
This would be like 5 LOC to add 😀

Reply:
In this specific place, yes, but an E2E working system also requires a corresponding way to communicate the endpoint to the client (probably as an extra `script` tag arg), probably via some new JavaScript "boot with specific config" API.